server/etcdserver/api/etcdhttp/health.go

```diff
@@ -99,30 +103,71 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(ctx context.Context, excludedAl
 }
 
 var (
+	// Deprecated: use rootHealthCheckCounter instead.
 	healthSuccess = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_success",
-		Help:      "The total number of successful health checks",
+		Help:      "The total number of successful health checks (Deprecated, use etcd_server_root_health_total instead)",
 	})
+	// Deprecated: use rootHealthCheckCounter instead.
 	healthFailed = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_failures",
-		Help:      "The total number of failed health checks",
+		Help:      "The total number of failed health checks (Deprecated, use etcd_server_root_health_total instead)",
 	})
+	rootHealthCheckGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "root_health",
+		Help:      "This metric records the result of the root readyz/livez check.",
+	},
+		[]string{"type"},
+	)
```

Comment:
Why do we need this root metric? Can't we just compute this from `healthCheckGauge`?

Reply:
The user would need to know that min of all healthCheckGauge gives you the overall health gauge. That might be inconvenient or error prone.

Reply:
It's pretty easy to do that in a prometheus query.. That's exactly what labels are meant for.

Reply:
ok. Removed rootHealthCheckGauge.